src/peergos/server/util/RateMonitor.java

```diff
@@ -0,0 +1,43 @@
+package peergos.server.util;
+
+import java.util.*;
+
+public class RateMonitor {
```

Comment:
Why not use a prometheus metric for this? Then it can  be  published and logged in grafana.

Reply:
If we want we can add a prometheus monitor on the requests as well, but prometheus can't do what this is doing.